### PR TITLE
Scope timeout to Ovh.Client, and allow each call method to set a specific one

### LINKED ISSUE #31: Scope timeout to Ovh.Client instance
Timeout param currently works in a strange way.
`Ovh.Client._httpClient` is static and instantiated at the first construction of an `Ovh.Client`.
During that first instantiation, `_httpClient.Timeout` is set to `Ovh.Client` constuctor's timeout param.

It means that during subsequent instantiations, the `timeout` param is effectively ignore.

### LINKED ISSUE #32: Allow to give a timeout per call
This would override the `Ovh.Client.Timeout` param for that specific call

### PR DESCRIPTION
**Pull request by Namaneo**
Fixes #31
Fixes #32

A timeout can be passed to the constructor of Ovh.Client, and will only stick to it. Under the hood, a cancellation token is prepared with it for each call. Alternatively, a specific timeout can be passed on a case-by-case basis on each async method (sync method left untouched as they are obsolete).

One constructor could be removed on Ovh.Client (both where almost the same at the end).

Just a word on the tests: currently, I can't find any solution to play with the timeout without requiring real HTTP calls. Since we are mocking the HttpMessageHandler, the CancellationToken is completely ignored. I'm open to any idea !

### CODE REVIEW

File: csharp-ovh/Client/Client.Put.cs [hunk 1]
```diff
@@ -68,9 +68,9 @@ public Task<string> PutAsync(string target, string data, bool needAuth = true)
         /// <param name="data">Json data to send as body</param>
         /// <param name="needAuth">If true, send authentication headers</param>
         /// <returns>API response deserialized to T by JSON.Net</returns>
```
Comment (LukeMarlin) [inline csharp-ovh/Client/Client.Put.cs:29]: Please add the documentation for the timeout param, stating that it overrides the client's one (and/or that it defaults to it)

**Review by LukeMarlin — CHANGES_REQUESTED** [review]

**Comment by LukeMarlin** [comment]
Thank you for your work, this is really appreciated. I've left a small comment for now, there might be more (for example regarding the testing issue) but it is bit late for that and i'm on my phone :)

File: csharp-ovh/Client/Client.Put.cs (hunk 1 above)
Comment (Namaneo) [reply csharp-ovh/Client/Client.Put.cs:29]: Documentation added on all Async calls

**Comment by Namaneo** [comment]
I just added a simple unit test checking that the CancellationToken is properly initialized and sent to the HttpMessageHandler. I had to update the FakeHttpMessageHandler to make it use the CancellationToken. I believe this test might be enough, as we don't necessarily want to check if a real HTTP request is cancelled (i.e. no framework testing).

Let me know if it's ok for you. :)

**Review by LukeMarlin — APPROVED** [review]

**Comment by LukeMarlin** [comment]
Looks good! Can you fix the conflicts so we can merge?

**Comment by Namaneo** [comment]
Fixed !

**Comment by LukeMarlin** [comment]
Thank you very much for the contribution!

**Pull request merged**